File: src/queues/queue_list.py
from queue_interface import QueueInterface
from src.list.node import Node

class QueueList(QueueInterface):
  """ implementation of a queue using a python list """

  def __init__(self):
    """ create an empty queue """
    self.list = []

  def isEmpty(self):
    """ check if queue is empty """
    return (self.list == [])

  def insert(self, cargo):
    """ add a new node to the queue """
    node = Node(cargo)
    self.list.append(node)

  def remove(self):
    """ remove and return the node at the top of the queue """
    return self.list.pop(0)

File: ruby/lib/queues/queue_list.rb
require_relative '../list/node'
require_relative './queue_interface'

class QueueList < QueueInterface

    attr_accessor :list

    def initialize
        @list = []
    end

    def empty?
        @list.empty?
    end

    def insert(cargo)
        node = Node.new(cargo)
        @list.push(node)
    end

    def remove
        @list.shift
    end

end